### PR TITLE
JetBrains: add basic integration with the local Cody app [WIP]

### LINKED ISSUE #53834: JetBrains: Let users conveniently connect to Cody App
We want to be able to connect to Cody App rather than sourcegraph.com or an enterprise instance to access the LLM, and everything else that we use the Sourcegraph API for.

Notes:
- The most important thing to remember is that the Cody App is like any enterprise instance in most aspects. The two key differences are that it's local and that we can count on the user to have admin access to it, which means we are free to use the full GraphQL API.
- We've got some context from #job-fair-app on determining whether App is installed, where to find the access token in the file system if it is installed, and how deep linking to App works.

A hopefully complete breakdown of what we'll want to do:
- [x] Create utility function to detect whether App is installed. Clue: [this](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/cody/src/services/LocalAppDetector.ts?L44) is how VS Code does it.
- [x] Create utility function to grab App's URL and access token from the file system (location and format are [here](https://github.com/sourcegraph/sourcegraph/pull/53782))
- [x] Add a third option to Settings, like [this](https://share.cleanshot.com/Hl0ncPV2).
  - If App is not detected, gray this out, and add a gray label below to explain that App is not present, with an App download link. Ideally, try to re-detect every time the Settings window gets focus, to auto-update this state
  - If App is detected then just explain in the label that the user needs no access token, we're ready to connect
- [x] Add "Download Cody App" and "Start Cody App" states to the sidebar (Like [this](https://share.cleanshot.com/WGYq9YzN) and [this](https://share.cleanshot.com/dmPk4N85) – First two screens from the [figma](https://www.figma.com/file/9112BsKsJc1BpO2j8XYwLL/%F0%9F%A4%96-Cody-VS-Code-%5BMain%5D?type=design&node-id=3288-142843&t=rG9PiIzT1awZjwqG-0)). The design is not important for the first iteration, just make it work
- [x] Modify ConfigUtil to work well if App is set as the current server. Its API works _exactly_ like an enterprise instance, so once we wire it up with ConfigUtil, it should just work!
- [x] Test it with App ([here](https://about.sourcegraph.com/app)) and see if it connects nicely
- Plus we have two follow-on tasks that I added to a separate issue for later: https://github.com/sourcegraph/sourcegraph/issues/53916

### PR DESCRIPTION
**Pull request by Gedochao**
~Fixes #53834~
Addresses most of #53834, the sidebar stuff + some other UX and other stuff will be added in a follow-up PR later.

This adds a third instance type for the local Cody App.
Only Macs are supported for now.
Make sure the app is running in the background when testing.

https://github.com/sourcegraph/sourcegraph/assets/18601388/88304210-d375-49d0-9c90-761efb844ff7

## Test plan
- some sanity-checks are included in the form of unit tests
- the local app setting should be greyed out if the app is not installed or if you're running on an unsupported platform (so not on Mac, for now)
- no access token should have to be set, the plugin should reuse the local app access token
- all API calls for Cody, including completions, chat and recipes should be done via the local app when enabled

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/cody/localapp/LocalAppManager.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      String responseBody = EntityUtils.toString(response.getEntity());
+      if (statusCode != 200) {
+        System.err.println(
+            "Could not fetch local Cody app version. Got status code "
+                + statusCode
+                + ": "
+                + responseBody);
+        return Optional.empty();
+      } else {
+        System.err.println("Running local Cody app version: " + responseBody);
```
Comment (szymonprz) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/localapp/LocalAppManager.java:95]: why error channel in case of success?

Comment (Gedochao) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/localapp/LocalAppManager.java:95]: ah, no reason.
Wasn't even sure I'd even keep these logs in the final version, I'll probably change to `System.out` for the ones I do.

**Review by vdavid — COMMENTED** [review]
Looks good to me so far!
It seems to cover all items from [the issue](https://github.com/sourcegraph/sourcegraph/issues/53834) except the chat UI part. If you want, you can open this for reviews, then merge it first and do the UI in a follow-up

File: client/jetbrains/src/main/java/com/sourcegraph/cody/localapp/LocalAppManager.java (hunk 1 above)
Comment (Gedochao) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/localapp/LocalAppManager.java]: Adjusted this part

**Comment by Gedochao** [comment]
I tweaked some things, rebased it on top of `main` and opened for review, so that it can get merged.
I will do a subsequent PR to fulfill the remaining requirements.

**Review by vdavid — APPROVED** [review]
Thanks for the cleanup. Looks good to merge to me!

**Pull request merged**